go-app-ussdapp.js

```diff
@@ -676,15 +827,108 @@ go.app = function() {
 
         // interstitial
         self.add("state_check_quiz_status", function(name) {
-            if (go.utils_project.has_untaken_quizzes()) {
-                return self.states.create("state_start_quiz");
-            } else {
-                return self.states.create("state_end_quiz_status");
-            }
+            return go.utils_project
+                .get_untaken_quizzes(self.im)
+                .then(function(untaken_quizzes) {
+                    if (untaken_quizzes.length > 0) {
+                        // get random quiz to take
+                        var quiz_to_take = go.utils_project.to_randomize_quizzes(self.im)
+                            ? untaken_quizzes[Math.floor(Math.random() * untaken_quizzes.length)]
+                            : untaken_quizzes[0];
+
+                        go.utils_project.init_quiz_status(self.im, quiz_to_take.id);
+
+                        return self.states.create("state_get_quiz_questions", quiz_to_take.id);
+                    } else {
+                        return self.states.create("state_end_quiz_status");
+                    }
+                });
+
         });
 
-        self.add("state_start_quiz", function(name) {
-            return self.states.create("state_end_quiz");
+        self.add("state_get_quiz_questions", function(name, quiz_id) {
+            return go.utils_project
+                .get_quiz(self.im, quiz_id)
+                .then(function(quiz) {
+                    // creates a random line-up of questions
+                    var random_questions = go.utils_project.to_randomize_questions(self.im)
```

Comment:
assuming they exist is fine. Having Required Configuration is a justifiable expectation. Reminds me, please add example one to a `config` folder same as other projects.